src/Evolution/Ringdown/MinimumAhCExcisionRadius.cpp

```diff
@@ -0,0 +1,344 @@
+                              TranslationMapOptions>{};
+
+  const domain::creators::sphere::TimeDependentMapOptions
+      ringdown_time_dependent_map_options{match_time,
+                                          shape_map_options,
+                                          rotation_map_options,
+                                          expansion_map_options,
+                                          translation_map_options,
+                                          true};
+
+  const double ahc_average_radius = ahc_inertial_at_match_time.average_radius();
+  const std::array<double, 3> ahc_ringdown_center =
+      trans_func_and_2_derivs.has_value()
+          ? trans_func_and_2_derivs.value()[0]
+          : std::array<double, 3>{0.0, 0.0, 0.0};
+  double ringdown_excision_factor = 0.94;
+  // These are assumed to be the x-positions of the excisions A/B at the start
+  // of the inspiral
+  const double mass_ratio = abs(excision_B_center[0] / excision_A_center[0]);
+  const double eps = 1e-3 / mass_ratio;
+  bool outer_converged = false;
+  size_t current_outer_iteration = 0;
+  const size_t max_iterations = 10;
+  size_t current_l_max = 20;
+
+  // The outer loop checks that multiple l_max values for the AhA/AhB excisions
+  // fit inside the ringdown domain excision
+  while (not outer_converged and current_outer_iteration < max_iterations) {
+    // Section for constructing strahlkorpers for AhA/AhB
+    const ylm::Strahlkorper<Frame::Grid> excision_a_inspiral_grid(
+        current_l_max, excision_A_radius, excision_A_center);
+    const ylm::Strahlkorper<Frame::Grid> excision_b_inspiral_grid(
+        current_l_max, excision_B_radius, excision_B_center);
+
+    const size_t points_size =
+        get<0>(ylm::cartesian_coords(excision_a_inspiral_grid)).size();
+    tnsr::I<DataVector, 3, Frame::Inertial> excision_a_inspiral_inertial_points{
+        points_size};
+    tnsr::I<DataVector, 3, Frame::Inertial> excision_b_inspiral_inertial_points{
+        points_size};
+    coords_to_different_frame(
+        make_not_null(&excision_a_inspiral_inertial_points),
+        ylm::cartesian_coords(excision_a_inspiral_grid), inspiral_domain,
+        inspiral_functions_of_time, match_time);
+    coords_to_different_frame(
+        make_not_null(&excision_b_inspiral_inertial_points),
+        ylm::cartesian_coords(excision_b_inspiral_grid), inspiral_domain,
+        inspiral_functions_of_time, match_time);
+
+    // Loop section finding the correct rmin_fac
+    double previous_ringdown_excision_factor = ringdown_excision_factor;
+    size_t current_inner_iteration = 0;
+    bool inner_converged = false;
+
+    // The inner loop changes the excision radius of the ringdown domain until
+    // the previous ringdown radius and current are within a set tolerance
+    while (not inner_converged and current_inner_iteration < max_iterations) {
+      const domain::creators::Sphere ringdown_rmin_domain_creator{
```

Comment:
Please add a detailed comment (at least 1 paragraph maybe more) of the assumptions being made in this DomainCreator and what it is being used for, etc.